src/Microsoft.AspNet.Mvc.Xml/DelegatingEnumerator.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.Xml
+{
+    public class DelegatingEnumerator<TWrappedOrDeclared, TDeclared> : IEnumerator<TWrappedOrDeclared>
+    {
+        private readonly IEnumerator<TDeclared> _inner;
+        private readonly IWrapperProvider _wrapperProvider;
+
+        public DelegatingEnumerator([NotNull] IEnumerator<TDeclared> inner, IWrapperProvider wrapperProvider)
+        {
+            _inner = inner;
+            _wrapperProvider = wrapperProvider;
+        }
+
+        public TWrappedOrDeclared Current
+        {
+            get
+            {
+                object obj = _inner.Current;
```

Comment:
(please take a look at the latest iteration) I cannot put a `var` here as the value of Current here would be of type `TWrapped ` and the cast would fail in the later line...like we would be trying to case TDeclared to TWrapped...